Dependencies for `SOC_FLASH_MCUX` are incorrect on LPC55xxx platforms

**Describe the bug**
Samples enabling `CONFIG_FLASH` fail to build on the LPC55S69 CPU1 target, due to the IAP flash driver being enabled when it should not be.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
   ` lpcxpresso55s69/lpc55s69/cpu1`
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
   Yes, the introduction of HWMv2

**To Reproduce**
<!--
Steps to reproduce the behavior:
1. west build -p -b lpcxpresso55s69/lpc55s69/cpu1 tests/subsys/secure_storage/psa/its -T secure_storage.psa.its.secure_storage.custom.store
2. See error
-->

**Expected behavior**
Build should pass

**Impact**
Blocking CI on #72102

**Logs and console output**
```
 /__w/zephyr/zephyr/drivers/flash/soc_flash_mcux.c:41:2: error: #error No matching compatible for soc_flash_mcux.c
   41 | #error No matching compatible for soc_flash_mcux.c
      |  ^~~~~
/__w/zephyr/zephyr/drivers/flash/soc_flash_mcux.c:53:10: fatal error: fsl_flash.h: No such file or directory
   53 | #include "fsl_flash.h"
      |          ^~~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```
**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA or Version used: d7ab4f25e13f975a9c3 (main)

--- a/soc/nxp/lpc/lpc55xxx/Kconfig.defconfig
+++ b/soc/nxp/lpc/lpc55xxx/Kconfig.defconfig
@@ -53,7 +53,7 @@ config CAN_MCUX_MCAN
 
 endif # SOC_LPC55S16
 
-if SOC_LPC55S69
+if SOC_LPC55S69_CPU0
 
 config SOC_FLASH_MCUX
 	default y
@@ -63,7 +63,7 @@ config SOC_FLASH_MCUX
 config I2S_MCUX_FLEXCOMM
 	select INIT_PLL0
 
-endif # SOC_LPC55S69
+endif # SOC_LPC55S69_CPU0
 
 if SOC_LPC55S69_CPU1
 

